service/bookturks/quiz/QuizTools.py

```diff
@@ -238,3 +239,21 @@ def unlink_all_tags_of_quiz(quiz):
         """
         for tag in quiz.quiztag_set.all():
             QuizTagAdapter.unlink_quiz(tag_name=tag.tag_name, quiz_id=quiz.quiz_id)
+
+    @staticmethod
+    def check_attempt_eligibility(self, request, quiz_complete_model, quiz_id):
+        """
+        Checks if the user still has number of attempts left for this quiz
+        :return:
+        """
+        attempted = 0
+        user_profile_model = request.session.get(USER_PROFILE_MODEL)
+        attempted_quiz = user_profile_model.attempted_quiz
+        for quiz_result in attempted_quiz:
+            if quiz_result.quiz_model.quiz_id == quiz_id:
+                attempted = quiz_result.attempts
```

Comment:
You can move the below lines (257-259) here and make the check here rather than breaking from the for loop.
Hence we can reduce the scope of the attempted local variable too.